app/src/test/java/org/kiwix/kiwixmobile/utils/NetworkUtilsTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Kiwix Android
+ * Copyright (C) 2018  Kiwix <android.kiwix.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.kiwix.kiwixmobile.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.Build;
+import android.util.Log;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kiwix.kiwixmobile.R;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NetworkUtilsTest {
+
+  @Mock private Context context;
+  @Mock private ConnectivityManager connectivity;
+  @Mock private NetworkInfo networkInfo;
+  @Mock private NetworkInfo networkInfo1;
+  @Mock private NetworkInfo networkInfo2;
+
+  public void testNetworkAvailability() {
+    NetworkInfo[] info = {networkInfo1,networkInfo2};
+    when(context.getSystemService(Context.CONNECTIVITY_SERVICE)).thenReturn(connectivity);
+    when(connectivity.getAllNetworkInfo()).thenReturn(info);
+
+    //one network is connected
+    when(networkInfo1.getState()).thenReturn(NetworkInfo.State.CONNECTED);
+    when(networkInfo2.getState()).thenReturn(NetworkInfo.State.DISCONNECTING);
+    assertEquals("",true,NetworkUtils.isNetworkAvailable(context));
```

Comment:
There are a couple places where there are no spaces after commas.

Reply:
@mhutti1 , I have fixed this and updated the branch.

Reply:
@mhutti1 , the Travis build is failing and I have no idea why. It's working perfectly on my local machine.

Reply:
@mhutti1 , The build passed on its own, after restarting it right now. So I think its all good.